File: Day 6/UnConvert/app/src/main/java/com/example/unconvert/MainActivity.java
package com.example.unconvert;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private Button button;
    private TextView textView;
    private EditText editTextText;

    private TextView textView2;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        button = findViewById(R.id.button);
        textView = findViewById(R.id.textView);
        editTextText = findViewById(R.id.editTextText);
        textView2 = findViewById(R.id.textView2);


        //there are two ways using setonclicklistener and by using onclick attribure

//        button.setOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View v) {
//
//                Toast.makeText(MainActivity.this, "Sucess", Toast.LENGTH_SHORT).show();
//                String s = editTextText.getText().toString();
//                int n = Integer.parseInt(s);
//
//                double result = n*2.205;
//
//                textView2.setText(result + " pounds");
//
//
//
//
//
//            }
//        });
    }


        public void calculator(View mayur)
        {
            String s = editTextText.getText().toString();
            int n = Integer.parseInt(s);

            double result = n*2.205;

            textView2.setText(result + " pounds");
        }
    }

File: Siddhi mam Tasks/Day 40/Firebase projects/Google_Sign/app/src/main/java/com/example/google_sign/MainActivity.kt
package com.example.google_sign

import android.app.Activity
import android.content.ContentValues.TAG
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.activity.result.contract.ActivityResultContract
import androidx.activity.result.contract.ActivityResultContracts
import androidx.credentials.CredentialManager
import androidx.credentials.CustomCredential
import androidx.credentials.GetCredentialRequest
import androidx.credentials.GetCredentialResponse
import androidx.credentials.PasswordCredential
import androidx.credentials.PublicKeyCredential
import com.example.google_sign.Constants.DEFAULT_WEB_CLIENT_ID

import com.example.google_sign.databinding.ActivityMainBinding
import com.google.android.gms.auth.api.identity.BeginSignInRequest
import com.google.android.gms.auth.api.signin.GoogleSignIn
import com.google.android.gms.auth.api.signin.GoogleSignInAccount

import com.google.android.gms.auth.api.signin.GoogleSignInClient
import com.google.android.gms.auth.api.signin.GoogleSignInOptions
import com.google.android.libraries.identity.googleid.GetGoogleIdOption
import com.google.android.libraries.identity.googleid.GoogleIdTokenCredential
import com.google.android.libraries.identity.googleid.GoogleIdTokenParsingException
import com.google.firebase.Firebase
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.GoogleAuthProvider
import com.google.firebase.auth.auth
import kotlinx.coroutines.CoroutineScope

class MainActivity : AppCompatActivity() {

    private lateinit var binding : ActivityMainBinding
    private lateinit var googleSignInClient : GoogleSignInClient
    private lateinit var auth:FirebaseAuth



    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        supportActionBar?.hide()







        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
            .requestIdToken(DEFAULT_WEB_CLIENT_ID) // this we have used coz our json was not working properly . there was problem of import otherwise our line will look like below
//           .requestIdToken(getString(R.string.default_web_client_id))
            .requestEmail()
            .build()

        googleSignInClient = GoogleSignIn.getClient(this,gso)
        auth = Firebase.auth

        binding.buttonSignInWithGoogle.setOnClickListener {
            val signInClient = googleSignInClient.signInIntent
            launcher.launch(signInClient)
        }

    }

    private val launcher = registerForActivityResult(ActivityResultContracts.StartActivityForResult())
    {
            result ->

        if(result.resultCode== Activity.RESULT_OK)
        {
            val task = GoogleSignIn.getSignedInAccountFromIntent(result.data)

            if(task.isSuccessful)
            {
                val account : GoogleSignInAccount?=task.result
                val credential = GoogleAuthProvider.getCredential(account?.idToken,null)
                auth.signInWithCredential(credential).addOnCompleteListener{
                    if(it.isSuccessful)
                    {
                        Toast.makeText(this,"done",Toast.LENGTH_SHORT).show()

                    }
                    else{

                        Toast.makeText(this,"failed",Toast.LENGTH_SHORT).show()

                    }
                }
            }

            else
            {
                Toast.makeText(this,"F",Toast.LENGTH_SHORT).show()
            }
        }
    }

}